src/reducers.js

```diff
@@ -300,19 +304,55 @@ function rasterSourceUUID(state = null, action) {
   };
 };
 
-function tasks(state = null, action) {
+function uploadFiles(state = null, action) {
   switch (action.type) {
-    case ADD_TASK:
-      const { uuid, filename, filesize } = action;
-      return {
-        ...state,
-        [action.uuid]: {
-          "uuid": uuid,
-          "filename": filename,
-          "size": filesize,
-          "status": 'Waiting'
-        }
-      };
+    case ADD_FILES_TO_QUEUE:
+      return action.files.map(file => {
+        return {
+          "name": file.name,
+          "size": file.size,
+          "uuid": null,
+          "status": 'WAITING'
+        };
+      });
```

Comment:
Could it be that this removes the existing files from the queue as well?

Reply:
Yes, currently, adding new files to the upload queue will remove the old ones. I thought the function `sendDataToLizardRecursive` will not work correctly in this case. However, I just changed the uploadFiles reducer and tested it. It worked like a charm. This is better than the old solution for sure. Thanks for your finding!